### PR TITLE
RELATED: RAIL-4899 remove numberjs from SDK

### PR DESCRIPTION
**Pull request by ivanmjartan**
JIRA: RAIL-4899

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by ivanmjartan** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15348/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16877/ : SUCCESS in 12m 12s

**Comment by ivanmjartan** [comment]
extended test - backstop

**Comment by ivanmjartan** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15350/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1276/ : SUCCESS in 26m 27s

**Comment by ivanmjartan** [comment]
extended test - tiger-cypress - integrated

File: libs/sdk-ui-charts/package.json [hunk 1]
```diff
@@ -73,8 +73,8 @@
         "validate-locales": "cd ../sdk-ui && npm run validate-locales",
         "validate-locales-ci": "cd ../sdk-ui && npm run validate-locales-ci"
     },
-    "dependencies": {
-        "@gooddata/numberjs": "^4.0.3",
+    "dependencies": {
```
Comment (kandl) [inline libs/sdk-ui-charts/package.json:76]: Unnecessary indentation

File: libs/sdk-ui-charts/src/charts/headline/internal/utils/HeadlineDataItemUtils.ts [hunk 2]
```diff
@@ -14,16 +14,20 @@ const DEFAULT_VALUE_WHEN_EMPTY = "–";
...
-    return colors2Object(formattedValue);
+    const parsedNumber = ClientFormatterFacade.convertValue(value);
+
+    // Based on the tests, when a format is not provided, we should refrain from formatting the value using the formatter, as the default format "#,##0.00" will be used.
+    // Additionally, the test necessitates that the value should remain unformatted.
+    if (!isEmpty(format)) {
+        return ClientFormatterFacade.formatValue(parsedNumber, format, separators);
+    }
+
+    if (parsedNumber === null || parsedNumber === undefined) {
```
Comment (kandl) [inline libs/sdk-ui-charts/src/charts/headline/internal/utils/HeadlineDataItemUtils.ts:26]: How is handled NaN? What use case is handled by the else block?

File: libs/sdk-ui-charts/tsconfig.json [hunk 3]
```diff
@@ -21,7 +21,7 @@
             "@gooddata/sdk-model": ["sdk-model/src"],
             "@gooddata/sdk-backend-spi": ["sdk-backend-spi/src"],
             "@gooddata/sdk-backend-mockingbird": ["sdk-backend-mockingbird/src"],
-            "@gooddata/util": ["util/src"]
+            "@gooddata/util": ["util/src"],
```
Comment (kandl) [inline libs/sdk-ui-charts/tsconfig.json:24]: comma probably should not be there (last line in the json block)

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15351/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1277/ : SUCCESS in 9m 06s

File: libs/sdk-ui-charts/package.json (hunk 1 above)
Comment (ivanmjartan) [reply libs/sdk-ui-charts/package.json:76]: fixed

File: libs/sdk-ui-charts/tsconfig.json (hunk 3 above)
Comment (ivanmjartan) [reply libs/sdk-ui-charts/tsconfig.json:24]: fixed

**Comment by ivanmjartan** [comment]
extended test - cypress - integrated

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3118/ : SUCCESS in 9m 23s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16881/ : SUCCESS in 12m 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/699/ : SUCCESS in 10m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4276/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/998/ : FAILURE in 1m 23s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/636/ : SUCCESS in 8m 58s

**Pull request merged**